### PR TITLE
[WEB-3583] Slider

### PR DESCRIPTION
**Pull request by aleksandar-r**
## Jira Ticket Link / Motivation

We find ourselves in need of a generic slider component, which will accept slides as children, and can be used on multiple places in Ably projects.

[WEB-3583](https://ably.atlassian.net/browse/WEB-3583)

## Summary of changes

Added React and Tailwind Slider component.

## How do you manually test this?

Check it out in Storybook

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

[WEB-3583]: https://ably.atlassian.net/browse/WEB-3583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core/Slider/Slider.stories.tsx [hunk 1]
```diff
@@ -1,45 +1,65 @@
-import React, { ReactNode } from "react";
+import React from "react";
 import Slider from "./component.tsx";
+import Icon from "../Icon/component.tsx";
```
Comment (jamiehenson) [inline src/core/Slider/Slider.stories.tsx:3]: These will cause merge conflicts now after the `set-up-14` merge, if you like I can resolve these

File: src/core/Slider/Slider.stories.tsx [hunk 2]
```diff
@@ -1,45 +1,65 @@
...
+              <p className="ui-text-p1 text-neutral-1300">{name}</p>
+              <p className="ui-text-p3 text-neutral-800">
+                Co-Founder & Technical Leader
+              </p>
+            </div>
+          </div>
+
+          <div className="w-[80px] h-1 sm:w-1 sm:h-full bg-neutral-500"></div>
+          <div className="flex items-center gap-4">
+            {/* <GatsbyImage /> */}
```
Comment (jamiehenson) [inline src/core/Slider/Slider.stories.tsx:29]: Could look into using a random avatar generator such as [this](https://randomuser.me/) to provide something to round out the profile, instead of leaving it blank. My favourite is [this](https://picsum.photos/) one which allows you to just plug in a URL into an `src` property instead of doing an async request. 

Could be a regular `img` or a `StaticImage` if done in the Gatsby way. This way we know how an image will look with little effort

File: src/core/Slider/component.tsx [hunk 3]
```diff
@@ -1,110 +1,109 @@
...
-  slideMaxWidth?: string;
-  mqEnableThreshold?: () => boolean;
+interface SliderProps {
+  children: ReactNode[];
+  interval?: number;
+}
 
-  container?: HTMLDivElement | null;
-};
+function SlideIndicator({
```
Comment (jamiehenson) [inline src/core/Slider/component.tsx:10]: Why use `function` syntax here instead of `const` like elsewhere?

File: src/core/Slider/component.tsx [hunk 4]
```diff
@@ -1,110 +1,109 @@
...
+  numSlides,
+  activeIndex,
+  interval,
+}: {
+  numSlides: number;
```
Comment (jamiehenson) [inline src/core/Slider/component.tsx:15]: Would be more consistent to have these types be their own interface at the top of the file like `SliderProps`

File: src/core/Slider/component.tsx [hunk 5]
```diff
@@ -1,110 +1,109 @@
...
+      {Array.from({ length: numSlides }, (_, i) => (
+        <div
+          key={i}
+          className={"relative w-40 h-4 mx-1 rounded-full bg-neutral-500"}
+        >
+          {i === activeIndex && (
+            <div
+              className="absolute inset-0 rounded-full bg-active-orange animate-[fillAnimation]"
+              style={{
+                animationDuration: `${interval}ms`,
```
Comment (jamiehenson) [inline src/core/Slider/component.tsx:30]: These inline properties might not be needed - it seems we can add to the `animate` Tailwind class above with further animation details: https://github.com/tailwindlabs/tailwindcss/discussions/9096#discussioncomment-3392101

**Review by jamiehenson — CHANGES_REQUESTED** [review]
Firstly, the new slider looks and feels amazing - nicely done :muscle:

As for the overall structure, I think we should find a way to merge the two versions of what we have here into one component - here’s my view for it. 

- the legacy `component.js` can continue to live in `/Slider` and will serve the one direct use case it’s used in
- two `component.css` files can be merged and live in `/Slider`
- for merging the two TSX components, the key differences here are the control positioning (the arrows) and the timed indicators. As such, we could introduce an `options` prop into the new `Slider` with the following properties,
	- `controlPosition?`: `”inline” | “floating”` (inline = old style, floating = new style)
	- `indicatorInterval?`: `number` (if present, show the newer timed indicators)
- in what is now `SliderOld`, the only props that are actually used across the codebase are `slides` (which could be `children` in the new Slider) and `mqEnableThreshold`, which can be in the options object mentioned above (if it's present, do the breakpoint thing the older one does)

Basically, I don’t think it would be too much of a stretch to replicate the functionality of the old slider in your cleaner newer version - all of the interactions are already there, it's just conditionally moving the position of the controls and indicators. The `component.js` version can stay as it is and just be used as a legacy component, and we remove its weird dependency inside the old Slider component.

File: src/core/Slider/component.tsx (hunk 5 above)
Comment (aleksandar-r) [reply src/core/Slider/component.tsx:30]: Tried it, did not really went well, but yeah, I can give it one more try

File: src/core/Slider/Slider.stories.tsx (hunk 1 above)
Comment (aleksandar-r) [reply src/core/Slider/Slider.stories.tsx]: Resolved them :grin:

File: src/core/Slider.tsx [hunk 6]
```diff
@@ -1,108 +1,190 @@
...
-  slideMinWidth?: string;
-  slideMaxWidth?: string;
-  mqEnableThreshold?: () => boolean;
+interface SliderProps {
+  children: ReactNode[];
+  options?: {
+    interval?: number;
+    controlPosition?: "inline" | "floating";
+    intervalIndicator?: boolean;
+    mqEnableThreshold?: () => boolean;
```
Comment (jamiehenson) [inline src/core/Slider.tsx:11]: Can kill this property now it's not being carried over

File: src/core/Slider/Slider.stories.tsx [hunk 7]
```diff
@@ -1,45 +1,89 @@
...
-    mqEnableThreshold: () => !window.matchMedia("(min-width: 48rem)").matches,
+    options: {
+      interval: 10000,
+      intervalIndicator: false,
+      controlPosition: "floating",
+    },
   },
 };
+
+export const Default = {};
```
Comment (jamiehenson) [inline src/core/Slider/Slider.stories.tsx:89]: Loading up Storybook, I think it makes the most sense to have this one first (Storybook respects export order) and called something like "Floating Control Position"

File: src/core/Slider.tsx [hunk 8]
```diff
@@ -1,108 +1,190 @@
...
+    >
+      {Array.from({ length: numSlides }, (_, i) =>
+        intervalIndicator ? (
+          <li
+            key={i}
+            className={"relative w-40 h-4 mx-1 rounded-full bg-neutral-500"}
+          >
+            {i === activeIndex && (
+              <li
+                className={`absolute inset-0 rounded-full bg-active-orange`}
```
Comment (jamiehenson) [inline src/core/Slider.tsx:44]: Don't need a template literal here as there's no interpolation

**Review by jamiehenson — COMMENTED** [review]
Looks really good! Nice work decommissioning that old component. Love the hover animation on the indicators.

I think though before we proceed the "inline" indicators should be made smaller to resemble the size of the old component indicators ([a reference](https://ably.github.io/ably-ui/?path=/docs/components-slider--docs)), with a small margin added between it and the slides. 

We don't even have to change the icon/animation I reckon, I think it's a strength that the slider has consistency there, but this should help to minimise any migration problems. The "old" Slider is used in Voltaire, so we'll be encountering this very soon.

**Review by jamiehenson — APPROVED** [review]
🙌 🙌 🙌
